activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -566,6 +575,18 @@ def local_database?(db_config)
         def schema_sha1(file)
           Digest::SHA1.hexdigest(File.read(file))
         end
+
+        def structure_dump_flags_for(adapter)
+          return structure_dump_flags[adapter.to_sym] if structure_dump_flags.is_a?(Hash)
+
+          structure_dump_flags
+        end
+
+        def structure_load_flags_for(adapter)
+          return structure_load_flags[adapter.to_sym] if structure_load_flags.is_a?(Hash)
+
+          structure_load_flags
+        end
```

Comment:
Can you make these two methods `structure_dump_flags_for` and `structure_load_flags_for`  private? Then we can assume the `adapter` is a symbol because it's not user provided. The API for database configurations will turn the adapter into a symbol from the configuration hash.

I'd then prefer the code to be:

```ruby
def structure_load_flags_for(adapter)
  if structure_load_flags.is_a?(Hash)
    structure_load_flags[adapter]
  else
   structure_load_flags
  end
end
```

Reply:
It looks like `adapter` comes as a string for tests so keeping the `to_sym` here, let me know what do you think